src/server/middleware/StaticAssetHandler.ts

```diff
@@ -49,11 +52,18 @@ export class StaticAssetHandler extends HttpHandler {
     const paths = Object.keys(this.mappings)
       .sort((pathA, pathB): number => pathB.length - pathA.length);
 
-    // Collect regular expressions for files and folders separately
+    // Collect regular expressions for files and folders separately.
+    // An initial value is needed in the array to prevent matching everything if these are empty.
     const files = [ '.^' ];
     const folders = [ '.^' ];
     for (const path of paths) {
-      (path.endsWith('/') ? folders : files).push(escapeStringRegexp(path));
+      const filePath = this.mappings[path];
+      if (filePath.endsWith('/') && !path.endsWith('/')) {
+        throw new InternalServerError(
+          `Static folders can only be used if both the URL and file path end on a slash, found ${path} and ${filePath}`,
```

Comment:
This needs some rephrasing to clarify what is meant by `found` and (optimally) what the user can do to correct the error condition.